client/src/redux/sagas/buildSetSaga.js

```diff
@@ -1,16 +1,61 @@
-import { takeEvery, call, put } from 'redux-saga/effects';
-import { BUILD_DATA } from '../constants';
-import { fetchBuildSets } from '../api';
-import { setBuildSets, setBuildSetsError } from '../actions';
+import { takeEvery, call, put, select } from 'redux-saga/effects';
+import { COMMITS } from '../constants';
+import { fetchBuildSets, fetchBuildReq, fetchBuilds } from '../api';
+import {
+ setBuildSets,
+ setBuildSetsError,
+ setBsID,
+ setBuilds
+} from '../actions';
+
+const commitSha = state => state.commits;
+function buildStr(build) {
+ let str = build
+  .map(build => 'buildrequestid__contains=' + build.buildrequestid)
+  .join('&');
+ return str;
+}
+
+function matchSha(commits, buildData) {
+ let cs = commits.map(commit => {
+  return commit.sha;
+ });
+ var filterBs = [];
+ cs.forEach(val => {
+  filterBs.push(
+   buildData
+    .map(obj => {
+     return obj;
+    })
+    .filter(item => item.sourcestamps[0].revision === val)
+  );
+ });
+ var merged = [].concat.apply([], filterBs);
+ console.log(merged);
+ let str = merged.map(bsid => 'buildsetid__contains=' + bsid.bsid).join('&');
+ return str;
+}
+
 function* handleBuildsLoad() {
  try {
   const buildSets = yield call(fetchBuildSets);
   yield put(setBuildSets(buildSets));
+  const commit = yield select(commitSha);
+  var buildReqStr = matchSha(commit, buildSets);
+  if (buildReqStr) {
```

Comment:
Why `var`? It is rarely used these days, when we have `let` and `const`
I'd rather check `if (buildReqStr.length > 0)` or something similar

Reply:
few var I need as I want them to use throughout the function, rest I have removed

Reply:
But const is fine here